app/Models/Support.php

```diff
@@ -213,5 +213,76 @@ public static function promoteDelegatesToDirect($topicNum, $nickNames)
 
         return;
     }
-    
+
+    public static function getAllSupporters($topic,$camp,$excludeNickID){
+        $nickNametoExclude = [$excludeNickID];
+        $support = self::where('topic_num','=',$topic)->where('camp_num','=',$camp)
+                ->where('end','=',0)
+                ->where('nick_name_id','!=',$excludeNickID)
+                ->where('delegate_nick_name_id',0)->groupBy('nick_name_id')->get(); 
+        $camp = Camp::where('camp_num','=',$camp)->where('topic_num','=',$topic)->first();
+        $allChildren = Camp::getAllChildCamps($camp);
+        $supportCount = 0;
+        if(sizeof($support) > 0 || count($support) >0){
+            foreach($support as $sp){
+                array_push( $nickNametoExclude, $sp->nick_name_id);
+            }
+        }
+        if(sizeof($allChildren) > 0 ){
+        foreach($allChildren as $campnum){
+            $supportData = self::where('topic_num',$topic)->where('camp_num',$campnum)->whereNotIn('nick_name_id',$nickNametoExclude)->where('delegate_nick_name_id',0)->where('end','=',0)->orderBy('support_order','ASC')->get();
+            if(count($supportData) > 0){
+                    foreach($supportData as $sp){
+                        array_push($nickNametoExclude, $sp->nick_name_id);
+                    }
+                    $supportCount = $supportCount + count($supportData);
+                }
+            }
+        }
+        return count($support)+$supportCount;
+    }
+
+    public static function ifIamSingleSupporter($topic_num,$camp_num=0,$userNicknames) {
```

Comment:
Not a big issue but just seems to be identical, please move the curly braces in the next line.

Reply:
It's okay in conditions but PHP called it a best practice and we are following it in almost all places.